File: src/my_cpp_pkg/src/smartphone.cpp
#include <stdio.h>
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

// Creating C++ Node with OOP
class SmartphoneNode: public rclcpp::Node
{
    public:
        SmartphoneNode(): Node("smartphone")
        {
            subscriber_ = this->create_subscription<example_interfaces::msg::String>(
                "robot_news", 10,
                std::bind(&SmartphoneNode::callbackRobotNews, this, std::placeholders::_1)
            );
            RCLCPP_INFO(this->get_logger(), "Smartphone has been started!");
        };
    
    private:

        void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg)
        {
            RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
        }

        rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;

};

int main(int argc, char **argv){

    // Initialize ROS2 communication
    rclcpp::init(argc, argv);

    // Object created into Shared pointer 
    // auto node = std::make_shared<rclcpp::Node>("cpp_test");

    // RCLCPP_INFO(node->get_logger(), "Hello CPP Node");

    auto node = std::make_shared<SmartphoneNode>();

    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

# inheriting from Node class
class SmartphoneNode(Node):

    def __init__(self):
        super().__init__("smartphone")
        self.subscriber_ = self.create_subscription(
            String, "robot_news", self.callback_robot_news, 10
        )
        self.get_logger().info("Smartphone has been started")

    def callback_robot_news(self, msg):
        self.get_logger().info(msg.data)
def main(args=None):
    rclpy.init(args=args)  # starts ROS2 communication and other features
    node = SmartphoneNode()

    rclpy.spin(node)  # node will be spinning and program wont stop
    rclpy.shutdown() # shutting down opened ros2 communication

if __name__ == "__main__":
    main()